src/cascadia/TerminalSettingsModel/ApplicationState.idl

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.Terminal.Settings.Model
+{
+    [default_interface] runtimeclass ApplicationState {
+        static ApplicationState GetForCurrentApp();
+
+        void Reload();
+        void Reset();
+        void Commit();
+
+        Boolean CloseAllTabsWarningDismissed;
+        Boolean LargePasteWarningDismissed;
+        Boolean MultiLinePasteWarningDismissed;

```

Comment:
(For a later PR), should we remove support for the existing warning-related settings? Or are they just gonna stick around as legacy settings?